File: solutions/lua/rna-transcription/1/rna-transcription.lua
return function(dna)
  local transcriptions = {
    ["G"] = "C",
    ["C"] = "G",
    ["T"] = "A",
    ["A"] = "U"
  }

  local transcribed = ""

  for i = 1, #dna do
    local nucleotide = dna:sub(i, i)
    transcribed = transcribed .. transcriptions[nucleotide]
  end

  return transcribed
end

File: solutions/ruby/rna-transcription/1/rna_transcription.rb
class Complement
  TRANSCRIPTIONS = {
    G: "C",
    C: "G",
    T: "A",
    A: "U"
  }

  def self.of_dna(sequence)
    sequence
      .to_s
      .upcase
      .chars
      .map { |c| transcribe(c) }
      .join
  end

  def self.transcribe(nucleotide)
    key = nucleotide.to_sym
    return "" if !TRANSCRIPTIONS.key?(key)
    TRANSCRIPTIONS[key]
  end
end
